lib/open_feature/sdk/api.rb

```diff
@@ -42,10 +42,12 @@ def configure(&block)
         block.call(configuration)
       end
 
-      def build_client(name: nil, version: nil)
-        client_options = Metadata.new(name: name, version: version).freeze
-        provider = Provider::NoOpProvider.new if provider.nil?
-        Client.new(provider: provider, client_options: client_options, context: context)
+      def build_client(version: nil, domain: nil)
+        client_options = Metadata.new(version: version, domain: domain).freeze
+
+        active_provider = provider(domain:).nil? ? Provider::NoOpProvider.new : provider(domain:)
+
+        Client.new(provider: active_provider, client_options: client_options, context: context)
```

Comment:
suggestion(non-blocking): If we use hash omission, we should be consistent and do it everywhere. 

```suggestion
        client_options = Metadata.new(version, domain:).freeze

        active_provider = provider(domain:).nil? ? Provider::NoOpProvider.new : provider(domain:)

        Client.new(provider: active_provider, client_options:, context:)
```